timescaledb/tutorials/analyze-bitcoin-blockchain/analyze-blockchain.md

```diff
@@ -0,0 +1,389 @@
+FROM one_hour_blocks b
+INNER JOIN coinbase c ON c.block_id = b.block_id
+GROUP BY bucket
+ORDER BY 1;
+```
+![Line graph with two lines showing the average fee and block reward, over the last five days](https://assets.timescale.com/docs/images/tutorials/bitcoin-blockchain/revenue_ratio.png)
+
+This chart analyzes the last five days of average miner revenue. The left
+axis shows the percentage of total revenue that
+comes from transaction fees (green) and block rewards (yellow). Most miner
+revenue actually comes from block rewards
+(6.25&nbsp;BTC at the moment). Fees never accounted for more than 3% in the
+last five days. 
+
+This kind of analysis can start discussions around the long-term fading of
+block rewards and how on-chain fees need to rise to
+incentivize miners and sustain the network. (Note that the left axis is
+logarithmic-scale, so it’s easier to see the green "fees" portion.)
+
+### How does block weight affect miner fees?
+You’ve already seen that more transactions in a block mean it’s more expensive
+to mine. Is it the same with block weights? The more transactions a block has,
+the larger its size (or weight), so the block weight and mining fee should
+be tightly correlated. 
+
+This query uses a 12-hour moving average to calculate the block weight and
+block mining fee over time.
+
+```sql
+WITH stats AS (
+   SELECT
+       bucket,
+       stats_agg(block_weight, block_fee_sat) AS block_stats
+   FROM one_hour_blocks
+   WHERE bucket > NOW() - INTERVAL '5 days'
+   GROUP BY bucket
+)
+SELECT
+   bucket as "time",
+   average_y(rolling(block_stats) OVER (ORDER BY bucket RANGE '12 hours' PRECEDING)) AS "block weight",
+   average_x(rolling(block_stats) OVER (ORDER BY bucket RANGE '12 hours' PRECEDING))*0.00000001 AS "mining fee"
+FROM stats
+ORDER BY 1
+```
+
+![block weight and fees](https://assets.timescale.com/docs/images/tutorials/bitcoin-blockchain/weight_fees.png)
+
+You can see that the block weight and block mining fee are indeed tightly
+connected to each other. In practice, you can also see that four million
+weight units are the size limit introduced in the 2017’s SegWit update.
+This means that, looking at this graph, there’s still room to grow for
+individual blocks, and they could include even more transactions.
+
+### What’s the average miner revenue per block?
+Now, analyze how much revenue miners actually generate by mining a
+new block on the blockchain, including fees and block rewards. This query
+analyzes the last day with 12-hour moving averages.
+
+```sql
+SELECT
+   bucket as "time",
+   average_y(rolling(stats_miner_revenue) OVER (ORDER BY bucket RANGE '12 hours' PRECEDING))*0.00000001 AS "revenue in BTC"
+FROM one_hour_coinbase
+WHERE bucket > NOW() - INTERVAL '1 day'
+ORDER BY 1
+```
+![Average miner revenue per block, plotted over the last day](https://assets.timescale.com/docs/images/tutorials/bitcoin-blockchain/miner_revenue_per_block.png)
+(THIS CHART NEEDS AN UPDATE - larger time frame?)
```

Comment:
don't know if you're ready to merge but if you are just a friendly reminder that this is here :)

Reply:
Thx I missed that, the chart is fine just needed to fix the SQL code